src/download.ts

```diff
@@ -136,7 +136,7 @@ export function getSkylinkUrl(this: SkynetClient, skylinkUrl: string, customOpti
   let url;
   if (opts.subdomain) {
     // Get the path from the skylink.
-    const skylinkPath = parseSkylink(skylinkUrl, { onlyPath: true });
+    const skylinkPath = parseSkylink(skylinkUrl, { onlyPath: true }) ?? "";
```

Comment:
Would it make sense to build this directly into `parseSkylink`? I mean the defaulting to an empty string instead of a `nil`/`undefined`.

Reply:
`null` seems to be a standard way to indicate "not found", for example this is how `String.prototype.match()` works

Reply:
This `parseSkylink(skylinkUrl, { onlyPath: true }) ?? ""` will not work because `??` only assigns the value on the right if the value on the left is `undefined` while `parseSkylink` never returns `undefined`.

Reply:
MDN:

> The nullish coalescing operator (??) is a logical operator that returns its right-hand side operand when its left-hand side operand is null or undefined, and otherwise returns its left-hand side operand.

I think it's correct as written, I can't do your suggestion because the null check is for a different call to parseSkylink

Reply:
Oh you're right then, sorry. Anyway, you should be able to move that code below that check and you won't need to check for nulls.

Reply:
We can't do that or the strict-null-check lint will complain

Reply:
Just FYI, this is resolved on my end.